### ISSUE TITLE
Programmatically Purge All 

### ISSUE THREAD

**Issue by kwcjr** [litespeedtech/lscache_wp#480]
I have another plugin that runs a few tasks after updating and I am trying to add these methods as well - however when I check the Headers before & after, I can't seam to get it to state 'x-litespeed-cache: miss' like when using the Toolbox Buttons...

`
if (class_exists('LiteSpeed\Purge')) {
  \LiteSpeed\Purge::purge_all();
}

//or 

if (class_exists('LiteSpeed\Purge')) {
  do_action('litespeed_purge_all');
}
`

**Comment by qtwrk** [comment]
what happens if you simply create a php file with code 

`<?php
require( './wp-load.php' );
if ( defined('LSCWP_V')) {
  do_action( 'litespeed_purge_all' );
  echo 'done';
}`

access by browser ,  does it purge ?    please verify by plugin's debug log

**Comment by kwcjr** [comment]
Doing the above works as intended; Purge Log is generated and the headers do state 'x-litespeed-cache: miss' after a refresh.

Once I try from within my plugin the; Purge Log is NOT generated and the headers still state 'x-litespeed-cache: hit' after a refresh. 
Browser cache is turned off & LS 'Serve Stale Setting' is set to OFF. 
13 other Cache Plugin Methods are accessible from my plugin environment as well so it should be that wp-load.php isn't loaded - however I tried that as well there was no change. 

Debug Logs show the Ajax POST coming in & LS does return the following logs:

08/29/22 00:37:02.595 [199.38.85.53:60256 1 WbN] 🧹 added *
08/29/22 00:37:02.595 [199.38.85.53:60256 1 WbN] 🧹 X-LiteSpeed-Purge: public,7d9_
08/29/22 00:37:02.595 [199.38.85.53:60256 1 WbN] 🧹 ❌ Bypassed cssjs delete as header sent (lscache purge after this point will fail)
08/29/22 00:37:02.595 [199.38.85.53:60256 1 WbN] 🧹 added _LOCALRES
08/29/22 00:37:02.595 [199.38.85.53:60256 1 WbN] 🧹 X-LiteSpeed-Purge: public,7d9_
08/29/22 00:37:02.596 [199.38.85.53:60256 1 WbN] [Object] flush!
08/29/22 00:37:02.596 [199.38.85.53:60256 1 WbN] 🧹 Flushed object cache
08/29/22 00:37:02.596 [199.38.85.53:60256 1 WbN] 🧹 Failed to reset opcode cache due to opcache not enabled
08/29/22 00:37:02.596 [199.38.85.53:60256 1 WbN] 🧹 Purge all

**Comment by qtwrk** [comment]
if standalone code works , that means the API is good , was that line even executed in your code  ?

**Comment by kwcjr** [comment]
Yes it's executed within my code and sent back to MainWP Dashboard - and it also shows up in the Debug Logs but doesn't generate Purge Logs or act as if it actually purged.

The plugin that I am referring to is MainWP Child - which is open source and includes a call to this method after any updates are performed on that site. https://github.com/mainwp/mainwp-child/blob/ca74d5a868ce61adf572c4bea31379bbcce47c90/class/class-mainwp-child-cache-purge.php#L631

This file calls the method above.
https://github.com/mainwp/mainwp-child/blob/master/class/class-mainwp-child-updates.php

**Comment by qtwrk** [comment]
is it like you have a main site , and then you are trying to purge on other site(s) that linked/controlled by mainWP ?

if this so , try check the server log , to see if it actually fired `x-litespeed-purge` header on remote site.
